add packet host driver

crunchywelch opened the pull request:
Signed-off-by: Aaron Welch <welch@packet.net>

ehazlett commented on drivers/packet/packet.go:
```diff
@@ -0,0 +1,392 @@
+		time.Sleep(10 * time.Second)
+	}
+
+	log.Infof("Provision time: %v.\n", time.Since(t0))
```
This should probably be debug.

ehazlett commented on drivers/packet/packet.go:
```diff
@@ -0,0 +1,392 @@
+		return err
+	}
+
+	switch d.OperatingSystem {
```
Why is this updating?  I'm not sure we want to force updates on users.  IMO this should be handled either in the base image or as a post-install step from the user (i.e. `docker-machine ssh machine -- sudo apt-get -y upgrade`).

ehazlett commented:
Overall this looks pretty good.  I have to test it.  You will also need to update the integration tests -- we merged a test refactor (should make it easier) yesterday.

Thanks!

crunchywelch commented:
Ok great, I updated the log statements to debug and removed the package updates. Will you let me know what I need to do with the integration tests, or will it be obvious when I merge?

Thanks!

crunchywelch commented:
hey @ehazlett just wanted to do a bump on this, are there changes I need to make to have this work with the next release? lmk what I can do to help get this in. Thanks!

ehazlett commented:
Hi , thanks for your efforts and persistence in submitting this driver. We are extremely excited that there is so much interest in Docker Machine and we really appreciate your interest. However, at this time it is proving to be extremely difficult for us to keep up with reviewing and testing each of these drivers for inclusion in the Machine core. We really want to switch to a more pluggable model, as well as polish up a few things about the driver model which need to be changed to ensure a smooth and sustainable future.

Therefore, we will be moving to a plugin model for 0.5 and would love to have you involved in the design and development process. We are closing the outstanding driver PRs at this time, but please keep the code. We will stick closely to the current driver interface and you should be able to re-use a lot (if not all) of the existing driver along with the new plugin model. We will be moving all of the drivers which are merged directly into Machine today to the plugin model when it is available, so there will be no special treatment of those, and there will be documentation outlining the process of developing and using a Docker Machine driver plugin.

With all of that being said, we want to apologize for the lack of feedback on your pull request. As contributors ourselves, we understand that being left in limbo is no fun. We would have liked to address this sooner, and in the future we will be more responsive around these kinds of issues.

Once again, we thank you for the contribution and the tremendous support. Keep hacking strong!

If you want to contribute to the design of the plugin model, we'd love to get your input on this issue where we will be planning it:

https://github.com/docker/machine/issues/1626

@ehazlett & @nathanleclaire
